### DISCUSSION TITLE
.3ddose and .egsphant to DICOM RT

### DISCUSSION THREAD

**Discussion by bsarey** [nrc-cnrc/EGSnrc#1233] [Questions]
Hi Community, 

is there a way to convert .3ddose and .egsphant files into DICOM RT files other than using 3D Slicer?

Thanks

**Comment by rtownson** [comment]
We don't provide any tools for this. If you do some programming, you can overwrite the doses in an RD dose file with the dose from the 3ddose file, and update the RP plan with the new voxel dimensions, etc. In python you can use pydicom to read and write DICOM files.

**Comment by crcrewso** [comment]
Not sure, completely untested, but this might help
https://github.com/christopherpoole/3DDose
Alternately:
https://discourse.slicer.org/t/converting-dosxyz-output-3ddose-to-dicom-rt-to-be-read-in-pinncale-and-eclipse/14499

**Comment by rtownson** [comment]
I think there's also some tools built into CERR using matlab for this.

**Comment by karl63** [comment]
As RTownson writes , it can be done with little programming skills in the Python programming language with the Pydicom package or in C++ with the DCMTK Toolkit from https://dicom.offis.de.
The trick is, that you do not start writing the DICOM file from scratch, but use a dicom dose file, that your treatment planning system (eg. Eclipse) has produced, as a template.
You open this existing file with Pydicom or DCMTK and change the relevant DICOM Tags for dose, resolution and dimensions according to the values you get from the .3ddose file.
Clearly to do so you should have some basic knowledge of the structure of the 3ddose file, as detailed in the EGSnrc documentation, and of a dicom dose file. I did such calculations 7 years or so myself, using DCMTK. I read that one can use the 3 D slicer or CERR as well but I did not try them out myself (did not use CERR because it uses mathlab, which is non free)
karl

Sent with [Proton Mail](https://proton.me/mail/home) secure email.
